src/main/java/com/contentgrid/gateway/runtime/routing/DefaultRuntimeRequestRouter.java

```diff
@@ -0,0 +1,38 @@
+package com.contentgrid.gateway.runtime.routing;
+
+import com.contentgrid.gateway.runtime.application.ServiceCatalog;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@RequiredArgsConstructor
+public class DefaultRuntimeRequestRouter implements RuntimeRequestRouter {
+
+    private final ServiceCatalog serviceCatalog;
+    private final RuntimeVirtualHostResolver virtualHostResolver;
+    private final RuntimeServiceInstanceSelector serviceInstanceSelector;
+
+    @Override
+    public Mono<ServiceInstance> route(ServerWebExchange exchange) {
+        return Mono.just(exchange.getRequest().getURI())
+                .map(this.virtualHostResolver::resolve)
+                .flatMap(Mono::justOrEmpty)
+                .switchIfEmpty(Mono.defer(() -> {
+                    log.debug("Could not resolve {} to app-id", exchange.getRequest().getURI());
+                    return Mono.empty();
+                }))
+                .map(this.serviceCatalog::findByApplicationId)
+                .map(services -> this.serviceInstanceSelector.selectService(exchange, services))
+                .doOnNext(result -> result.ifPresentOrElse(
+                        service -> log.debug("Routing request {} to {}://{}:{}{}", exchange.getRequest().getURI(),
```

Comment:
Wouldn't it be sufficient to log the request host and service host + port? All the res on that line is pretty much just noise.